src/main/java/com/park/easyrecruit/servlet/application/Applications.java

```diff
@@ -21,28 +21,26 @@
  * @author andrei
  */
 @WebServlet(name = "Applications", urlPatterns = {"/Applications"})
-@ServletSecurity(value = @HttpConstraint(rolesAllowed = {"ClientRole"}))
+@ServletSecurity(value = @HttpConstraint(rolesAllowed = {"ClientRole","ManageInterviewRole"}))
 public class Applications extends HttpServlet {
 
     @Inject
     private ApplicationBean applicationBean;
 
-    /**
-     * Handles the HTTP <code>GET</code> method.
-     *
-     * @param request servlet request
-     * @param response servlet response
-     * @throws ServletException if a servlet-specific error occurs
-     * @throws IOException if an I/O error occurs
-     */
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response)
             throws ServletException, IOException {
-        
+
         // anyone GET => applications of current user (My Applications)
         // HR GET + positionId => list all applications for specified position
+        String action = (String) request.getParameter("action");
+        if(action != null) {
+            applicationBean.executeAction(action,
+                    Integer.parseInt((String) request.getParameter("positionId")),
+                    (String) request.getParameter("username"));
+        }
         
-        request.setAttribute("applications", applicationBean.getMany(request.getUserPrincipal().getName()));
-        request.getRequestDispatcher("/WEB-INF/pages/applications.jsp").forward(request, response);
+        request.setAttribute("applicationsAll", applicationBean.getAll());
```

Comment:
This servlet serves the `My Applications` page. It should not serve all applications and a GET method should never change data. Please update.
Also I see that `master` already has a `ApplicationsListAll` servlet.